src/shell/shell.c

```diff
@@ -488,6 +495,27 @@ static int shell_barrier (flux_shell_t *shell, const char *name)
 
 static int shell_init (flux_shell_t *shell)
 {
+    bool required = false;
+    const char *rcfile = shell_conf_get ("shell_initrc");
+
+    /*  If initrc was passed on command line, it is an error if
+     *   the file is not found. O/w treat missing initrc as empty file
+     */
+    if (optparse_getopt (shell->p, "initrc", &rcfile) > 0)
+        required = true;
+    else
+        rcfile = shell_conf_get ("shell_initrc");
+
+    if (shell->verbose)
+        log_msg ("Loading %s", rcfile);
+
+    if ((shell_rc (shell, rcfile) < 0) && (errno != ENOENT || required)) {
```

Comment:
Might need to audit `shell_rc(()` to ensure errno is not spuriously set on success or set incorrectly on failure, by passing through from another non-fatal error.

Reply:
Good point, though `errno` should not be checked on success (if I am that is a bug). I'll carefully review `shell_rc()` to ensure it only return `ENOENT` on missing file.

Actually, would it be better to just `stat()` the file before calling `shell_rc()` to make the missing file case more explicit?

  I went back and forth on error handling for missing initrc files during development. It seemed like the rules should be

 * it is not an error if the main initrc doesn't exist, unless `--initrc FILE` was used.
 * it is not an erorr if `source` with a glob pattern has no matches, e.g. `source *.lua` shouldn't fail when there are no matching files.
 * `source` of a non-pattern *should* fail if the file doesn't exist. (Thus the unfortunate need to add the function `source_if_exists`

Reply:
Oh right, sorry - misspoke on the success case.

Agreed, those rules LGTM.

> Actually, would it be better to just `stat()` the file before calling `shell_rc()` to make the missing file case more explicit?

Or `access(3)`.  No opinion here, either way works IMHO.

Reply:
Yeah, I switched to `access` in my working copy. Will add some basic tests before pushing again.